web-ui/src/app/components/tree-visualizer/tree-visualizer.js

```diff
@@ -459,17 +460,18 @@ var jQuery = require('jquery');
                             right: $window.width() - treeRect.right,
                             bottom: $window.innerHeight() - treeRect.bottom,
                         };
+
                     // The maximum to the right the tooltip can be positioned without overlapping the viewport
-                    var rightestLeft = parseInt($window.width() - tooltip.outerWidth() - 20, 10)
+                    var rightestLeft = parseInt($window.width() - tooltip.outerWidth() - 20, 10);
+                    // var rightestLeft = parseInt(Math.max($window.width() - tooltip.outerWidth() - 20, targetRect.right), 10);
                     var tooltipV = {
-                        left: Math.max(10, Math.min(rightestLeft, parseInt(targetRect.left + (targetRect.width / 2) - (tooltip.outerWidth() / 2) + 8, 10))) + "px",
+                        left: Math.max(20, Math.min(rightestLeft, parseInt(targetRect.left + (targetRect.width / 2) - (tooltip.outerWidth() / 2) + 8, 10))) + "px",
                         top: parseInt(targetRect.top - tooltip.outerHeight() - 24, 10)
-                    }
+                    };
                     // Speech bubble arrow positioned independently from tooltip rectangle
                     var arrowV = {
-                        left: parseInt(targetRect.left + (targetRect.width / 2) - (arrowRect.width / 2), 10), //the centre of the target node
-                        top: parseInt(tooltipV.top + tooltip.outerHeight() - 1) //the bottom of the tooltip balloon
-                    }
+                        left: Math.min(Math.max((targetRect.left - parseInt(tooltipV.left, 10) - 8), 0), tooltip.width())
```

Comment:
`tooltip.width()` could be replaced here by `tooltip.width() - 24`: then the arrow won't scroll away from the tooltip when fading out.